guidance/library/_json_schema.py

```diff
@@ -0,0 +1,183 @@
+from typing import Any, Callable, Mapping, MutableMapping, Optional, Sequence
+
+import guidance
+from guidance.library import char_range, one_or_more, optional, zero_or_more
+
+from .._grammar import GrammarFunction, select
+
+
+@guidance(stateless=True)
+def _gen_json_int(lm):
+    return lm + optional("-") + one_or_more(char_range("0", "9"))
```

Comment:
First digit should not be zero unless the integer is identically zero / negative zero
```python
@guidance(stateless=True)
def _gen_json_int(lm):
    pos_nonzero = char_range("1", "9") + zero_or_more(char_range("0", "9"))
    return lm + optional("-") + select(["0", pos_nonzero])
```

Reply:
Fair point. I was assuming that no parsing library would actually care, but since you have provided the correction..... :-)

Reply:
Neither `json.load{,s}` nor the python interpreter will parse something with leading zeroes as an integer. I just thought better safe than sorry -- we really want to make sure these outputs are "loads-able" :)

Reply:
Now I've actually tried it.... indeed. I'm slightly surprised, since `int("00001")` works just fine.

Reply:
But curiously `json.loads("1.0e+023")` _does_ work

Reply:
From https://www.json.org/json-en.html -- looks like that's part of the spec:
![number](https://github.com/guidance-ai/guidance/assets/927410/b3e06ae3-7451-4ef8-ab17-06c3e2e292f4)